crates/core/src/metadata/mod.rs

```diff
@@ -286,6 +304,7 @@ struct MetadataInner {
     my_node_id: OnceLock<GenerationalNodeId>,
     nodes_config: Arc<ArcSwap<NodesConfiguration>>,
     partition_table: Arc<ArcSwap<PartitionTable>>,
+    scheduling_plan: Arc<ArcSwap<SchedulingPlan>>,
```

Comment:
Adding the `SchedulingPlan` to the `Metadata` means that we will quite aggressively sync it between nodes whenever it gets updated. The less information that we sync that way, the less is the risk to overload the system. Moreover, the information is not needed by all the cluster components. E.g. if we deploy a `LogServer` node only, then this node would also pay the price for exchanging the `SchedulingPlan` w/o requiring it.

Reply:
Got it, I understand the concern now! Thanks for the feedback!